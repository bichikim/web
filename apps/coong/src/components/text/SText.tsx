import {ComponentProps} from 'solid-js'
import {cva} from 'class-variance-authority'

const rootStyle = cva('', {
  variants: {
    size: {
      lg: 'text-lg py-.5 px-1.5 rd-1',
      md: 'text-base py-.5 px-1.5 rd-1',
      sm: 'text-sm py-.5 px-1.5 rd-1',
    },
  },
})

export interface STextProps extends ComponentProps<'span'> {
  size?: 'sm' | 'md' | 'lg'
}

export const SText = (props: STextProps) => {
  return <span {...props} class={rootStyle({class: props.class, size: props.size})} />
}
