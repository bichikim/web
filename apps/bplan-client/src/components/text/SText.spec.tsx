/**
 * @vitest-environment jsdom
 */
import {describe, expect, it} from 'vitest'
import {render} from '@solidjs/testing-library'
// import userEvent from '@testing-library/user-event'
import {SText} from './SText'

describe('SText', () => {
  it('should render', () => {
    const {getByText} = render(() => <SText>Hello</SText>)

    expect(getByText('Hello')).toBeInTheDocument()
    expect(getByText('Hello')).toMatchSnapshot()
  })

  it.each([
    //
    {size: 'md'},
    {size: 'lg'},
    {size: 'sm'},
  ])('should render size: $size', ({size}: any) => {
    const {getByText} = render(() => <SText size={size}>Hello</SText>)

    expect(getByText('Hello')).toMatchSnapshot()
  })
})
