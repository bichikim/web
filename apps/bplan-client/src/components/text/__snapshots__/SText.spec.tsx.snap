// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SText > should render 1`] = `
<span
  class=""
>
  Hello
</span>
`;

exports[`SText > should render size: 'lg' 1`] = `
<span
  class="text-lg py-.5 px-1.5 rd-1"
  size="lg"
>
  Hello
</span>
`;

exports[`SText > should render size: 'md' 1`] = `
<span
  class="text-base py-.5 px-1.5 rd-1"
  size="md"
>
  Hello
</span>
`;

exports[`SText > should render size: 'sm' 1`] = `
<span
  class="text-sm py-.5 px-1.5 rd-1"
  size="sm"
>
  Hello
</span>
`;
